plugins/cpu/fu-plugin-cpu.c

```diff
@@ -10,32 +10,64 @@
 #include "fu-hash.h"
 #include "fu-cpu-device.h"
 
+struct FuPluginData {
+	gboolean		 is_intel;
+	gboolean		 has_cet;
+};
+
 void
 fu_plugin_init (FuPlugin *plugin)
 {
 	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+	fu_plugin_alloc_data (plugin, sizeof (FuPluginData));
 }
 
 gboolean
 fu_plugin_coldplug (FuPlugin *plugin, GError **error)
 {
+	FuPluginData *data = fu_plugin_get_data (plugin);
 	gsize length;
-	g_autofree gchar *data = NULL;
+	g_autofree gchar *buf = NULL;
 	g_auto(GStrv) lines = NULL;
 
-	if (!g_file_get_contents ("/proc/cpuinfo", &data, &length, error))
+	if (!g_file_get_contents ("/proc/cpuinfo", &buf, &length, error))
 		return FALSE;
 
-	lines = g_strsplit (data, "\n\n", 0);
+	lines = g_strsplit (buf, "\n\n", 0);
 	for (guint i = 0; lines[i] != NULL; i++) {
 		g_autoptr(FuCpuDevice) dev = NULL;
 		if (strlen (lines[i]) == 0)
 			continue;
 		dev = fu_cpu_device_new (lines[i]);
 		if (!fu_device_setup (FU_DEVICE (dev), error))
 			return FALSE;
+		if (g_strcmp0 (fu_device_get_vendor (FU_DEVICE (dev)), "GenuineIntel") == 0) {
+			data->is_intel = TRUE;
+			if (fu_cpu_device_has_shstk (dev) &&
+			    fu_cpu_device_has_ibt (dev))
+				data->has_cet = TRUE;
+		}
 		fu_plugin_device_add (plugin, FU_DEVICE (dev));
 	}
 
 	return TRUE;
 }
+
+gboolean
+fu_plugin_add_security_attrs (FuPlugin *plugin, GPtrArray *attrs, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	if (data->is_intel) {
```

Comment:
So AMD will be having something as well (see https://community.amd.com/community/amd-business/blog/2019/10/21/amd-and-microsoft-secured-core-pc)

Reply:
Is it going to be in cpuids as well?

Reply:
I would hope it shows up there, but again until the kernel is actually exporting flags details are slim.